src/main.cpp

```diff
@@ -149,115 +159,126 @@ int main(int argc, char *argv[])
 
     ROS_INFO("Connecting to : %s @ %d Baud", SensorPort.c_str(), SensorBaudrate);
 
-    // Create a VnSensor object and connect to sensor
-    VnSensor vs;
-
-    // Default baudrate variable
-    int defaultBaudrate;
-    // Run through all of the acceptable baud rates until we are connected
-    // Looping in case someone has changed the default
-    bool baudSet = false;
-    while(!baudSet){
-        // Make this variable only accessible in the while loop
-        static int i = 0;
-        defaultBaudrate = vs.supportedBaudrates()[i];
-        ROS_INFO("Connecting with default at %d", defaultBaudrate);
-        // Default response was too low and retransmit time was too long by default.
-        // They would cause errors
-        vs.setResponseTimeoutMs(1000); // Wait for up to 1000 ms for response
-        vs.setRetransmitDelayMs(50);  // Retransmit every 50 ms
-
-        // Acceptable baud rates 9600, 19200, 38400, 57600, 128000, 115200, 230400, 460800, 921600
-        // Data sheet says 128000 is a valid baud rate. It doesn't work with the VN100 so it is excluded.
-        // All other values seem to work fine.
-        try{
-            // Connect to sensor at it's default rate
-            if(defaultBaudrate != 128000 && SensorBaudrate != 128000)
+    while (ros::ok())

```

Comment:
did you actually change anything in here or is github just being dumb?

Reply:
besides the while(ros::ok()) loop

Reply:
I ended up wrapping a huge block of code (all the connection and initialization steps) inside a new while() loop, which allows the node to re-connect properly in the event that the sensor is reset. But yeah, the diff looks really ugly.

Reply:
That's what I figured but I was struggling to determine if there were other changes. Sounds good.